tidy3d/plugins/smatrix/smatrix.py

```diff
@@ -75,15 +75,15 @@ class ComponentModeler(Tidy3dBaseModel):
         title="Folder Name",
         description="Name of the folder for the tasks on web.",
     )
-    element_mappings: Dict[Element, Dict[Element, Callable[[complex], complex]]] = pd.Field(
-        {},
+    element_mappings: Tuple[Tuple[Element, Element, float], ...] = pd.Field(
```

Comment:
Probably better `complex`, most generally maybe the element just has some phase offset.

Reply:
So, when i made it complex when doing these changes, for some reason it would throw validation errors
 if i supplied +1 or -1. I never really understood why that was (maybe it doesn't type cast the tuple elements themselves?). I could change it back and write a custom validator for it though?